templates/score.html

```diff
@@ -0,0 +1,26 @@
+<p>Results are here:</p>
+<!-- Parādām mūsu faila rindas -->
+<UL  id="scores">
+{% for value in lines %}
+<li>
+    {{ value }}
+</li>
+
+{% endfor %}
+</UL>
+<table>
+    <tbody id="tabula"></tbody>
+</table>
+ <script>
+     dati = document.getElementById('scores').innerText
+         let test = JSON.stringify(dati)
+        console.log(test)
+        var objekts = JSON.parse(test)
+        console.log(objekts)
+let tbody = document.getElementById("tabula")
```

Comment:
Bet, ja gribās paeksperimentēt un tomēr izvadīt teksta failu, piemēram, tabulas veidā, kas šeit ir darīts, to var izdarīt īsāk.
Piemēram:
```js
dati = document.getElementById('scores').innerText
rindas = dati.split('\n')
let tabula = document.getElementById('tabula')
for(let i=0; i<rindas.length; i++) {
  kolonnas = rindas[i].split(' ')
  tabula.innerHTML += `<tr>
    <td>${kolonnas[0]}</td>
    <td>${kolonnas[1]}</td>
    <td>${kolonnas[2]}</td>
  </tr>`
}
```
JSON stringify un parse neko nedeva, jo sākotnēji nebija JSON objekts.
Dotās funkcijas var izmantot, piemēram, ja mums ir šāds te JSON objekts:
```js
jsonObjekts: {
 jsonObjektaAtribūts: "Atribūts"
}
```
Tad mēs šo varam ar stringify konvertēt simbolu virknē, aizsūtīt vai saņemt, un tad pārkonvertēt atpakaļ JSON objektā.
Taču, ja mēs ar kaut kādu simbolu virkni: "Simbolu virkne" mēģināsim izpildīt šīs te funkcijas, tad tās neko nedarīs.

Lai nu kā, domu gājiens bija pareizajā virzienā.
Es gribēju parādīt, ka var konkatenēt simbolu virknes bez + simbola, vienkārši ar interpolāciju, kur ērti izmanto tildes simbolus `Šeit ir simbolu virkne ${mainīgais}`. Man liekas, tā ir ērtāk.

Par pašiem datiem domāju, ka ir skaidrs. Mums ir fails, kas ir simbolu virkne, kur kolonnas atdalītas ar " " atstarpēm un rindas ar "\n" jaunām rindām. Tad sākumā mēs visu faila simbolu virkni sadalām pēc "\n" simbola, kur iegūstam masīvu no visām rindām, pēc tam katru rindu sadalām pēc " " tukšuma un iegūstam katras rindas kolonnu.

Reply:
Jā, es nedaudz nomaldījos, domāju tikt cauri pa vienkāršo - ielasīt mainīgajā JSON objektu un pēc tam no tā mainīgā ar cikla palīdzību vilkt laukā atslēgu un vērtību, to attēlojot tabulas šūnā, taču, kad katrs simbols drukājās atsevišķi, nesapratu, ka faila saturs nav kā JSON objekts. Tagad man pieleca!

Reply:
Jā, bet tas ir lieks darbs. HTML failā serveris jau ir atsūtījis datus. Ja gribās tā darīt, tad servera pusē ir jāizveido korekts JSON objekts un jāpadod uz doto html skatu kā mainīgo un tad to var darīt tur. Bet es domāju, ka dotā uzdevuma ietvaros tas nav vajadzīgs. Vieglāk ir to visu kodēt ar pliku html. Vispār labā prakse izvairīties no lieka JS. Vienīgi var tad tikai JS kodēt, piemēram, React JS, vai Vue JS - tas ir tam paredzēts.